cdn/signUrl/SignUrlSample/Program.cs

```diff
@@ -0,0 +1,119 @@
+            {
+                return 1;
+            }
+
+            Console.WriteLine($"Signed URL: {signedUrl}");
+            return 0;
+        }
+
+        // [START example]
+        /// <summary>
+        /// Creates signed URL for Google Cloud SDN
+        /// More details about order of operations is here: <see cref="https://cloud-dot-devsite.googleplex.com/cdn/docs/signed-urls#creatingkeys"/>
+        /// </summary>
+        /// <param name="url">The Url to sign. This URL can't include Expires and KeyName query parameters in it</param>
+        /// <param name="keyName">The name of the key used to sign the URL</param>
+        /// <param name="encodedKey">The key used to sign the Url</param>
+        /// <param name="expirationTime">Expiration time of the signature</param>
+        /// <returns>Signed Url that is valid until {expirationTime}</returns>
+        public static string CreateSignedUrl(string url, string keyName, string encodedKey, DateTime expirationTime)
+        {
+            var builder = new UriBuilder(url);
+
+            long unixTimestampExpiration = ToUnixTime(expirationTime);
+
+            char queryParam = string.IsNullOrEmpty(builder.Query) ? '?' : '&';
+            builder.Query += $"{queryParam}Expires={unixTimestampExpiration}&KeyName={keyName}".ToString();
+
+            // Key is passed as base64url encoded
+            byte[] decodedKey = Base64UrlDecode(encodedKey);
+
+            // Computes HMAC SHA-1 hash of the URL using the key
+            byte[] hash = ComputeHash(decodedKey, builder.Uri.AbsoluteUri);
+            string encodedHash = Base64UrlEncode(hash);
+            
+            builder.Query += $"&Signature={encodedHash}";
+            return builder.Uri.AbsoluteUri;
+        }
+
+        private static long ToUnixTime(DateTime date)
+        {
+            var epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+            return Convert.ToInt64((date - epoch).TotalSeconds);
+        }
+
+        private static byte[] Base64UrlDecode(string arg)
+        {
+            string s = arg;
+            s = s.Replace('-', '+'); // 62nd char of encoding
+            s = s.Replace('_', '/'); // 63rd char of encoding
+
+            return Convert.FromBase64String(s); // Standard base64 decoder
+        }
+
+        private static string Base64UrlEncode(byte[] inputBytes)
+        {
+            var output = Convert.ToBase64String(inputBytes);
+
+            output = output.Replace('+', '-')      // 62nd char of encoding
+                           .Replace('/', '_');     // 63rd char of encoding
+
+            return output;
+        }
+
+        private static byte[] ComputeHash(byte[] secretKey, string signatureString)
+        {
+            var enc = Encoding.ASCII;
+            using (HMACSHA1 hmac = new HMACSHA1(secretKey))
+            {
+                hmac.Initialize();
+
+                byte[] buffer = enc.GetBytes(signatureString);
```

Comment:
also probably unnecessary declaration, you can put this inside `ComputeHash(...)`

Reply:
I could write the entire method in single line like this:
<b>return new HMACSHA1(secretKey).ComputeHash(Encoding.ASCII.GetBytes(signatureString));</b>
However I don't think it would be good sample this way.
People reading this may have different skill sets and the sample should be easy to follow, debug and understand to all of them. We are not in the competition on the smallest number of lines.
I prefer "unnecessary" declarations for better clarity vs. smaller number of lines of code - especially in sample code.

Reply:
SGTM